pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -339,34 +354,53 @@ func (s *Server) createIstioCA(client corev1.CoreV1Interface, opts *CAOptions) (
 		} else {
 			log.Info("Use local self-signed CA certificate")
 		}
-
 		// The cert corresponding to the key, self-signed or chain.
 		// rootCertFile will be added at the end, if present, to form 'rootCerts'.
 		signingCertFile := path.Join(LocalCertDir.Get(), "ca-cert.pem")
 		//
 		certChainFile := path.Join(LocalCertDir.Get(), "cert-chain.pem")
 		s.caBundlePath = certChainFile
-
 		caOpts, err = ca.NewPluggedCertIstioCAOptions(certChainFile, signingCertFile, signingKeyFile,
 			rootCertFile, workloadCertTTL.Get(), maxCertTTL, caRSAKeySize.Get())
 		if err != nil {
 			return nil, fmt.Errorf("failed to create an istiod CA: %v", err)
 		}
 	}
-
 	istioCA, err := ca.NewIstioCA(caOpts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create an istiod CA: %v", err)
 	}
-
 	// TODO: provide an endpoint returning all the roots. SDS can only pull a single root in current impl.
 	// ca.go saves or uses the secret, but also writes to the configmap "istio-security", under caTLSRootCert
-
 	// rootCertRotatorChan channel accepts signals to stop root cert rotator for
 	// self-signed CA.
 	rootCertRotatorChan := make(chan struct{})
 	// Start root cert rotator in a separate goroutine.
 	istioCA.Run(rootCertRotatorChan)
-
 	return istioCA, nil
 }
+
+// createExternalCA initializes the Istio RA signing functionality.
+// the caOptions defines the external provider
+func (s *Server) createExternalCA(client kubelib.Client,
+	opts *caOptions,
+	caCertFile string) (*ra.IstioRA, error) {
+
+	spiffe.SetTrustDomain(opts.TrustDomain)
+	maxCertTTL := maxWorkloadCertTTL.Get()
+	if SelfSignedCACertTTL.Get().Seconds() > maxCertTTL.Seconds() {
+		maxCertTTL = SelfSignedCACertTTL.Get()
+	}
+	if opts.CAType == KubernatesCA {
```

Comment:
What do you mean by KubernetesCA? Do you mean:
1) K8s CA signs the workload certs, or
2) External CA using the K8s API?

It seems you meant (2)? If so would you change the name to something like KubernetesAPICA?

Reply:
I mean External CA using the K8s CSR API (which may or may not be K8sCA). Does KubernatesExtCA work?